addons/Omen/Omen.lua

```diff
@@ -0,0 +1,287 @@
+windower.add_to_chat(161,"3: You have reduced your foe's HP by 5000 using a single magic attack without performing a magic burst.")
+windower.add_to_chat(161,"3: You have reduced your foe's HP by 14999 using a single magic attack without performing a magic burst.")
+windower.add_to_chat(161,"3: You have reduced your foe's HP by at least 15000 using a single magic attack without performing a magic burst.")
+coroutine.sleep(1)
+windower.add_to_chat(161,"1: Vanquish 6 foes.")
+windower.add_to_chat(161,"2: Use 30 weapon skills on your foes.")
+windower.add_to_chat(161,"3: Reduce your foe's HP by at least 15000 using a single magic attack without performing a magic burst.")
+windower.add_to_chat(161,"4: Use 12 abilities on your foes.")
+windower.add_to_chat(161,"5: Reduce your foe's HP by at least 2000 in a single auto-attack.")
+coroutine.sleep(1)
+windower.add_to_chat(161,"A spectral light flares up."..string.char(0x7f).."1")
+end
+
+
+windower.register_event('prerender', function()
+    if obj_time < 1 then return end
+    if obj_time ~= (end_time - os.time()) then
+        obj_time = end_time - os.time()
+        refresh()
+    end
+end)
+
+windower.register_event('zone change', function(zone)
+    texts.visible(IMAGE,false)
+    floor_obj = "Waiting for objectives..."
+    reset_objectives()
+    if zone == 292 then -- Henge
+        texts.visible(IMAGE,true)
+    end
+end)
+
+texts.visible(IMAGE,false)
+if windower.ffxi.get_info() and windower.ffxi.get_info().zone == 292 then
+texts.visible(IMAGE,true)
+end
+
+windower.register_event('incoming text', function(original, modified, mode)
+    if mode == 161 then -- Omen messages are 161 color, except total time extension messages which are 121 and irrelevant
+        if string.match(original,"^%d") then
+            for k,v in pairs (messages) do
+                if string.find(original,v.init) then
+                    if objectives[tonumber(original:match("%d"))].mes ~= tonumber(v.id) then -- New Objective
+                        objectives[tonumber(original:match("%d"))].amt = 0
+                    end
+                    objectives[tonumber(original:match("%d"))].mes = tonumber(v.id)
+                    objectives[tonumber(original:match("%d"))].req = tonumber(string.sub(original:match(v.check),1,-2))				
+                elseif string.find(original,v.eval) then
+                    objectives[tonumber(original:match("%d"))].amt = tonumber(string.sub(original:match(v.check),1,-2))
+                    if objectives[tonumber(original:match("%d"))].mes == 0 then -- if loading mid-floor
+                        objectives[tonumber(original:match("%d"))].mes = tonumber(v.id)
+                        objectives[tonumber(original:match("%d"))].req = -1
+                    end
+                end
+                refresh()
+            end
+        elseif string.find(original,"%d+ omen") then
+            omens = original:match("%d+")
```

Comment:
I don't know the format of the messages, but this could theoretically match a different number than the `if` check above. I'd restrict the `find` string some more and match the `match` string accordingly.